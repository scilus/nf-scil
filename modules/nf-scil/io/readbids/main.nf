process IO_READBIDS {
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.0.sif':
        'scilus/scilus:2.0.0' }"

    input:
        tuple path(bids_folder), path(fsfolder), path(bidsignore)

    output:
        path("tractoflow_bids_struct.json")             , emit: bidsstructure
        path "versions.yml"                             , emit: versions


    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def fsfolder = task.ext.fs_folder ? "--fs " + task.ext.fs_folder : ''
    def bidsignore = task.ext.bidsignore ? "--bids_ignore " + task.ext.bidsignore : ''
    def readout = task.ext.readout ? "--readout " + task.ext.readout : ""
    def clean_flag = task.ext.clean_bids ? '--clean ' : ''

    """
    scil_bids_validate.py $bids_folder tractoflow_bids_struct.json\
        $readout \
        $clean_flag\
        $fsfolder\
        $bidsignore\
        -v

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: 2.0.0
    END_VERSIONS
    """


    stub:
    def args = task.ext.args ?: ''
    """
    scil_bids_validate.py -h

    touch tractoflow_bids_struct.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: 2.0.0
    END_VERSIONS
    """
}
