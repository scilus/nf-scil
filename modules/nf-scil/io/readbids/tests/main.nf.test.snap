{
    "io - readbids - folder": {
        "content": [
            {
                "0": [
                    "tractoflow_bids_struct.json:md5,58e0494c51d30eb3494f7c9198986bb9"
                ],
                "1": [
                    "versions.yml:md5,0af9864a1818468702fa7d6ac7985e29"
                ],
                "bidsstructure": [
                    "tractoflow_bids_struct.json:md5,58e0494c51d30eb3494f7c9198986bb9"
                ],
                "versions": [
                    "versions.yml:md5,0af9864a1818468702fa7d6ac7985e29"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-22T14:18:58.455287"
    }
}